queue_services/entity-filer/src/entity_filer/filing_processors/correction.py

```diff
@@ -67,14 +67,17 @@ def process(correction_filing: Filing, filing: Dict, filing_meta: FilingMeta, bu
         )
     )
 
-    if business.legal_type in ['SP', 'GP', 'BC', 'BEN', 'CC', 'ULC'] or \
-            is_special_resolution_correction(filing, business, original_filing):
-        correct_business_data(business, correction_filing, filing, filing_meta)
-    else:
+    corrected_filing_type = filing['correction']['correctedFilingType']
+
+    if business.legal_type not in ['SP', 'GP', 'BC', 'BEN', 'CC', 'ULC'] and \
+            not is_special_resolution_correction(filing, business, original_filing) \
+            or corrected_filing_type == 'conversion':
         # set correction filing to PENDING_CORRECTION, for manual intervention
         # - include flag so that listener in Filing model does not change state automatically to COMPLETE
         correction_filing._status = Filing.Status.PENDING_CORRECTION.value  # pylint: disable=protected-access
         setattr(correction_filing, 'skip_status_listener', True)
+    else:
+        correct_business_data(business, correction_filing, filing, filing_meta)
```

Comment:
I think this is too fragile.  The `correct_business_data` logic should have a strict check to ensure business data doesn't accidentally get corrected.

I think I would keep the original code and add a `corrected_filing_type != conversion`.  It might make sense to `if` move the condition in the `if` block into its own function if it's getting too messy and hard to understand.

Reply:
I updated